android/src/main/java/nl/kega/reactnativeemdk/BarcodeScannerManager.java

```diff
@@ -27,118 +27,185 @@
 import java.util.Iterator;
 import java.util.List;
 
+/*
+ * This class exposes the following methods to the React Native application:
+ *  - init()
+ *  - read()
+ *  - scan()
+ *  - cancel()
+ *  - enable()
+ *  - disable()
+ */
 public class BarcodeScannerManager extends ReactContextBaseJavaModule implements LifecycleEventListener {
-    
+
     public final ReactApplicationContext context;
 
     private BarcodeScannerThread scannerthread = null;
 
+    /*
+     * Constructor
+     * The OS is checked for Zebra/Motorola Enterprise features before creating the scanner thread.

```

Comment:
Nothing functional happens if the check is false. The scanner thread is just never created and app runs, creates the GUI and all that, but can't make the calls to the scanner because it doesn't exist physically or software-wise.